src/main/java/org/generationcp/middleware/dao/ims/LotDAO.java

```diff
@@ -634,4 +639,96 @@ private void createLotRows(List<Lot> lots, Query query, boolean withReservationM
+		+ "  SUM(CASE WHEN transaction.trnstat = 1 AND transaction.trnqty <= 0 THEN transaction.trnqty * -1 ELSE 0 END) AS withdrawalTotal, " //
+		+ "  lot.comments as comments, " //
+		+ "  users.uname as createdByUsername, " //
+		+ "  MIN(transaction.trndate) as createdDate, " //
+		+ "  MAX(CASE WHEN transaction.trnstat = 0 AND transaction.trnqty > 0 THEN str_to_date(CAST(transaction.trndate as CHAR), '%Y%m%d') ELSE null END) AS lastDepositDate, " //
+		+ "  MAX(CASE WHEN transaction.trnstat = 1 AND transaction.trnqty <= 0 THEN str_to_date(CAST(transaction.trndate as CHAR), '%Y%m%d') ELSE null END) AS lastWithdrawalDate " //
+		+ "FROM ims_lot lot " //
+		+ "       LEFT JOIN ims_transaction transaction ON transaction.lotid = lot.lotid AND transaction.trnstat <> 9 " //
+		+ "       INNER JOIN germplsm g on g.gid = lot.eid " //
+		+ "       INNER JOIN names n ON n.gid = lot.eid AND n.nstat = 1 " //
+			// location id and scale id are suppossed to be mandatory but they are not, left join should be replaced by inner join
+		+ "       LEFT JOIN location l on l.locid = lot.locid " //
+		+ "       LEFT join cvterm scale on scale.cvterm_id = lot.scaleid " //
+		+ "       INNER JOIN workbench.users users on users.userid = lot.userid " //
+		+ "WHERE lot.etype = 'GERMPLSM' "; //
+
+	private String buildSearchLotsQuery(final LotsSearchDto lotsSearchDto) {
+		final StringBuilder query = new StringBuilder(SEARCH_LOT_QUERY);
+		if (lotsSearchDto != null) {
+			if (lotsSearchDto.getStatus() != null) {
+				query.append(" and lot.status = ").append(lotsSearchDto.getStatus());
+			}
+		}
+		query.append(" GROUP BY lot.lotid ");
+		return query.toString();
+	}
+
+	public List<LotDto> searchLots(final LotsSearchDto lotsSearchDto, final Pageable pageable) {
+		try {
+			final String filterLotsQuery = buildSearchLotsQuery(lotsSearchDto);
+			final SQLQuery query = this.getSession().createSQLQuery(filterLotsQuery);
+			query.addScalar("lotId");
+			query.addScalar("stockId");
+			query.addScalar("gid");
+			query.addScalar("mgid");
+			query.addScalar("designation");
+			query.addScalar("status");
+			query.addScalar("locationId");
+			query.addScalar("locationName");
+			query.addScalar("scaleId");
+			query.addScalar("scaleName");
+			query.addScalar("actualBalance");
+			query.addScalar("availableBalance");
+			query.addScalar("reservedTotal");
+			query.addScalar("withdrawalTotal");
+			query.addScalar("comments");
+			query.addScalar("createdByUsername");
+			query.addScalar("createdDate", Hibernate.DATE);
+			query.addScalar("lastDepositDate", Hibernate.DATE);
+			query.addScalar("lastWithdrawalDate",Hibernate.DATE);
+
+			query.setResultTransformer(Transformers.aliasToBean(LotDto.class));
+
+			if (pageable!= null) {
+				query.setFirstResult(pageable.getPageSize() * pageable.getPageNumber());
+				query.setMaxResults(pageable.getPageSize());
```

Comment:
To add **sorting**, there is this method org.generationcp.middleware.dao.SampleListDao#addOrder that I've been thinking we can move below org.generationcp.middleware.dao.GenericDAO#addOrder to work as generic _pageable-to-sql_ **orderBy** utility

cc @clarysabel

Reply:
👍